### PR TITLE
Add Systems > Compatibility doc subsection

### PR DESCRIPTION
**Pull request by mnishiguchi**
addresses https://github.com/nerves-project/nerves/issues/759

### CODE REVIEW

File: docs/Systems.md [hunk 1]
```diff
@@ -39,6 +39,31 @@ appropriate toolchain for that target. The system and toolchain also reference a
...
+The Nerves System (`nerves_system_*`) dependency determines the OTP version
+running on the target. It is possible that a recent update to the Nerves
+System pulled in a new version of Erlang/OTP. If you are using an official
+Nerves System, you can verify this by reviewing the chart below or
+`CHANGELOG.md` file that comes with the release.
+
+|            | rpi    | rpi0   | rpi2   | rpi3a  | rpi3   | rpi4   | bbb    | x86_64 | osd32mp1 |
+| ---------- | ------ | ------ | ------ | ------ | ------ | ------ | ------ | ------ | -------- |
+| OTP 25.0   | 1.19.0 | 1.19.0 | 1.19.0 | 1.19.0 | 1.19.0 | 1.19.0 | 2.14.0 | 1.19.0 | 0.10.0   |
+| OTP 24.3.2 | 1.18.4 | 1.18.4 | 1.18.4 | 1.18.4 | 1.18.4 | 1.18.4 | -      | 1.18.4 | 0.9.4    |
```
Comment (fhunleth) [inline docs/Systems.md:53]:
```suggestion
| OTP 24.3.2 | 1.18.4 | 1.18.4 | 1.18.4 | 1.18.4 | 1.18.4 | 1.18.4 | 2.13.4 | 1.18.4 | 0.9.4    |
```

This adds the BBB version that has OTP 24.3.2.

**Comment by mnishiguchi** [comment]
What about this?

```elixir
Mix.install [:req]

## Preprocess metadata per nerves_br version

nerves_br_versions = ~w[
 1.16.0
 1.16.1
 1.16.2
 1.16.3
 1.16.4
 1.16.5
 1.17.0
 1.17.1
 1.17.2
 1.17.3
 1.17.4
 1.18.0
 1.18.1
 1.18.2
 1.18.3
 1.18.4
 1.18.5
 1.18.6
 1.19.0
 1.19.1
 1.20.0
]

nerves_br_metadata =
  Enum.reduce(nerves_br_versions, %{}, fn nerves_br_version, acc ->
    %{status: 200, body: create_build_content} = Req.get!("https://raw.githubusercontent.com/nerves-project/nerves_system_br/v#{nerves_br_version}/create-build.sh")
    %{status: 200, body: tool_versions_content} = Req.get!("https://raw.githubusercontent.com/nerves-project/nerves_system_br/v#{nerves_br_version}/.tool-versions")

    acc
    |> Map.put(
      nerves_br_version,
      %{"nerves_br" => nerves_br_version}
      |> Map.merge(Regex.named_captures(~r/NERVES_BR_VERSION=(?<buildroot>[0-9.]*)/, create_build_content, include_captures: true))
      |> Map.merge(Regex.named_captures(~r/erlang (?<otp>[0-9.]*)/, tool_versions_content, include_captures: true))
    )
  end)

## Build metadata per nerves_br_target version

target_tags = [
  bbb: ~w[2.14.0 2.13.4 2.13.3 2.12.3],
  rpi: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  rpi0: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  rpi2: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  rpi3: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  rpi3a: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  rpi4: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  x86_64: ~w[1.19.0 1.18.4 1.18.3 1.17.3],
  osd32mp1: ~w[0.10.0 0.9.4 0.9.3 0.8.3],
]

Enum.reduce(target_tags, %{}, fn {target, tags}, acc ->
  tag_nerves_br_metadata =
    Enum.map(tags, fn tag ->
      %{status: 200, body: mix_lock_content} = Req.get!("https://raw.githubusercontent.com/nerves-project/nerves_system_#{target}/v#{tag}/mix.lock")
      nerves_br = Regex.named_captures(~r/:hex, :nerves_system_br, "(?<nerves_br>[0-9.]*)"/, mix_lock_content, include_captures: true) |> Access.fetch!("nerves_br")

      %{ tag => nerves_br_metadata |> Access.fetch!(nerves_br) }
    end)

  acc
  |> Map.put(target, tag_nerves_br_metadata)
end)
```

**Comment by fhunleth** [comment]
Sounds good to me. 

Were you planning on adding more `nerves_system_*` versions?

**Comment by fhunleth** [comment]
For where to put the script, you can put it in the `scripts` directory. For the chart, I'd keep it where you have it. I'm actually not sure which is better, a Mermaid chart or a plain markdown table like you have. 

If the Mermaid chart looks like the way to go, see https://github.com/nerves-networking/vintage_net/blob/960a092aeb78dbedbba9ddd80061c3fbaa882f2b/lib/vintage_net/connectivity/check_logic.ex#L13-L24 and https://github.com/nerves-networking/vintage_net/blob/215495533cb642eeb172daba08208a454f19b36f/mix.exs#L137-L159 for an example that works with ex_doc.

**Comment by mnishiguchi** [comment]
👍 I'll do experiments in https://github.com/mnishiguchi/nerves_systems_compatibility.

**Comment by fhunleth** [comment]
@mnishiguchi This seems useful enough to merge and then update in future PRs. What do you think?

**Comment by fhunleth** [comment]
Thanks!

**Pull request merged**